shared.go

```diff
@@ -91,20 +91,23 @@ func loadSharedCache() error {
 
 // saveSharedCache creates a temporary directory to share between garble processes.
 // This directory also includes the gob-encoded cache global.
-func saveSharedCache() (string, error) {
+func saveSharedCache() (err error) {
 	if sharedCache == nil {
 		panic("saving a missing cache?")
 	}
-	dir, err := os.MkdirTemp("", "garble-shared")
-	if err != nil {
-		return "", err
-	}
 
-	cachePath := filepath.Join(dir, "main-cache.gob")
-	if err := writeGobExclusive(cachePath, &sharedCache); err != nil {
-		return "", err
+	if sharedTempDir == "" {
+		sharedTempDir, err = os.MkdirTemp("", "garble-shared")
+		if err != nil {
+			return err
+		}
+
+		os.Setenv("GARBLE_SHARED", sharedTempDir)
```

Comment:
I assume these changes around GARBLE_SHARED are related to your older "preexec" changes which are now dropped. If so, please revert, because otherwise we're just making this code more complex for no reason.

Or if you actually need these changes, please explain, because I'm not following why they're happening.